### PR TITLE
User can create an entry

### PR DESCRIPTION
**Pull request by lauragracem**
Screen Shot 2022-05-31 at 2 33 09 PM
Screen Shot 2022-05-31 at 2 38 46 PM

https://user-images.githubusercontent.com/60913524/171288849-6f6d0a33-f8cd-4cdd-a70f-c230b48dfec9.mov

https://user-images.githubusercontent.com/60913524/171289226-57bd5a10-587f-4d77-a490-73c2f932b8d5.mov

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+var nextEntryId = 0;
```
Comment (BrettAlbright) [inline js/main.js]: this value should come from `data.nextEntryId`

File: js/main.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  var newEntry = {
+    title: $titleInput.value,
+    notes: $notesInput.value,
+    photoUrl: $photoInput.value,
+    id: nextEntryId
+  };
+  nextEntryId++;
+
+  var entryBase = [];
```
Comment (BrettAlbright) [inline js/main.js]: the entries should be unshifted into the `data.entries` array

File: js/main.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  var entryJSON = localStorage.getItem('journal-entries');
+  if (entryJSON !== null) {
+    entryBase = JSON.parse(entryJSON);
+  }
+
+  entryBase.push(newEntry);
+
+  var newEntryJSON = JSON.stringify(entryBase);
+  localStorage.setItem('journal-entries', newEntryJSON);
```
Comment (BrettAlbright) [inline js/main.js]: this should be in the `data.js` file and you should store the entire data object in local storage

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  $photoUpdate.src = 'images/placeholder-image-square.jpg';
+  $titleInput.value = '';
+  $notesInput.value = '';
+  $photoInput.value = '';
```
Comment (BrettAlbright) [inline js/main.js]: instead of manually resetting the inputs, I would call the reset method on the form itself

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
a few changes needed

File: js/main.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    notes: $notesInput.value,
+    photoUrl: $photoInput.value,
+    id: data.nextEntryId
+  };
+  data.nextEntryId++;
+
+  data.entries.unshift(newEntry);
+
+  var newEntryJSON = JSON.stringify(data);
+  localStorage.setItem('journal-entries', newEntryJSON);
```
Comment (BrettAlbright) [inline js/main.js]: you should not be using local storage here, the only time local storage is used is before unload.

**Review by Zhiwutian — CHANGES_REQUESTED** [review]
The data.js file is not present, and a little more localstorage removal from the submit callback

**Review by Zhiwutian — APPROVED** [review]
👍🏼

**Pull request merged**